pandas/core/internals/blocks.py

```diff
@@ -991,6 +991,18 @@ def setitem(self, indexer, value):
 
         # set
         else:
+            if (
+                self.is_object
+                and not is_ea_value
+                and arr_value.dtype.kind in ["m", "M"]
+            ):
+                # https://github.com/numpy/numpy/issues/12550
+                #  numpy will incorrect cast to int if we're not careful
+                if is_list_like(value):
+                    value = list(value)
```

Comment:
`np.asarray`?

Reply:
until numpy#12550 is fixed this is the only way i found to get this to work